src/coreclr/tools/Common/Compiler/TypeExtensions.cs

```diff
@@ -46,6 +46,25 @@ public static DefType GetClosestDefType(this TypeDesc type)
             return (DefType)type;
         }
 
+        public static MethodDesc GetConstructorForCreateInstanceIntrinsic(this TypeDesc type)
+        {
+            MethodDesc ctor = type.GetDefaultConstructor();
+            if (ctor == null)
+            {
+                MetadataType activatorType = type.Context.SystemModule.GetKnownType("System", "Activator");
+                if (type.IsValueType && type.GetParameterlessConstructor() == null)
+                {
+                    ctor = activatorType.GetKnownMethod("ValueTypeWithNoConstructorMethod", null);
```

Comment:
This looks too tightly coupled with corelib for it to live in in this file. Is there a better place for it? E.g. Common/TypeSystem/IL/Stubs/RuntimeHelpersIntrinsics.cs 

Also, this looks too special to be an extension method.